hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieCopyOnWriteTableInputFormat.java

```diff
@@ -247,81 +239,33 @@ private List<FileStatus> listStatusForSnapshotMode(JobConf job,
       boolean shouldIncludePendingCommits =
           HoodieHiveUtils.shouldIncludePendingCommits(job, tableMetaClient.getTableConfig().getTableName());
 
+      HiveHoodieTableFileIndex fileIndex =
+          new HiveHoodieTableFileIndex(
+              engineContext,
+              tableMetaClient,
+              props,
+              HoodieTableQueryType.SNAPSHOT,
+              partitionPaths,
+              queryCommitInstant,
+              shouldIncludePendingCommits);
+
+      Map<String, List<FileSlice>> partitionedFileSlices = fileIndex.listFileSlices();
+
       // NOTE: Fetching virtual key info is a costly operation as it needs to load the commit metadata.
       //       This is only needed for MOR realtime splits. Hence, for COW tables, this can be avoided.
       Option<HoodieVirtualKeyInfo> virtualKeyInfoOpt = tableMetaClient.getTableType().equals(COPY_ON_WRITE) ? Option.empty() : getHoodieVirtualKeyInfo(tableMetaClient);
```

Comment:
This is the main change. Earlier it used to be simply 
`Option<HoodieVirtualKeyInfo> virtualKeyInfoOpt = getHoodieVirtualKeyInfo(tableMetaClient);`.

Reply:
Great find @codope! Glad we're able to identify the root-cause of that slow-down

Reply:
I'd suggest we take this fix one step further -- instead of fetching virtual key-info here let's push it inside `createFileStatusUnchecked` by pass the meta-client in there